bin/predictions.py

```diff
@@ -60,12 +62,48 @@
     # Set up joblib memory to use for caching hard steps
     memory = Memory(config["memory_cache_location"], verbose=0)
 
-    # Load and filter the bulks
+    # Load the bulks
     bulks_delayed = dask.delayed(memory.cache(load_bulks))(
         config["input_options"]["bulk_file"]
     )
     bulk_bag = db.from_delayed([bulks_delayed])
     bulk_df = bulk_bag.to_dataframe().repartition(npartitions=50).persist()
+
+    # Create pourbaix lmdb if it doesnt exist
+    if "filter_by_pourbaix_stability" in list(config["bulk_filters"].keys()):
+        conditions = config["bulk_filters"]["filter_by_pourbaix_stability"]
+        if not os.path.isfile(conditions["lmdb_path"]):
+            warnings.warn(
+                "No lmdb was found here:"
+                + conditions["lmdb_path"]
+                + " Making the lmdb instead"
+            )
+            bulk_bag = bulk_bag.repartition(npartitions=200)
+            pbx_dicts = bulk_bag.map(
+                get_pourbaix_info, conditions["mp_api_key"]
+            ).compute()
+
+            db = lmdb.open(
```

Comment:
I would suggest moving lines 86-103 to a separate function (write_pourbaix_info?) somewhere so that bin/predictions is easier to read.